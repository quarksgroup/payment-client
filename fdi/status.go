package fdi

import (
	"context"

	"github.com/quarksgroup/payment-client/client"
)

type APIStatus struct {
	State string
}

func (c *Client) Status(ctx context.Context) (*APIStatus, *client.Response, error) {
	endpoint := "status"
	out := new(statusResponse)
	res, err := c.do(ctx, "GET", endpoint, nil, out, true)
	return convertStatus(out), res, err
}

type statusResponse struct {
	Status string `json:"status"`
	Data   struct {
		State string `json:"state"`
	} `json:"data"`
}

func convertStatus(res *statusResponse) *APIStatus {
	return &APIStatus{
		State: res.Data.State,
	}
}
