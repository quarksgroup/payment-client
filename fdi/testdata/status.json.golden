{
  "gref": "84F9CC90-28CA-4385-952A-9000EC5A11A8",
  "ref": "CD8FEB19-5AD0-4496-ADD6-36CCF1B67B71",
  "state": "processing"
}
